src/lib/u2.js

```diff
@@ -1437,3 +1437,52 @@ foam.CLASS({
     }
   ]
 });
+
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'CSS',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'code'
+    },
+    {
+      name: 'name',
+      factory: function() { return 'CSS-' + this.$UID; }
+    }
+  ],
+
+  methods: [
+    function installInClass(cls) {
+      // There might be several CSS axioms on each class.
+      // Each one will try to run this code.
+      // In order to prevent them all from running, we check if cls.create
+      // has the __cssHandled property set.
+      if ( cls.create.__cssHandled ) return;
+
+      var oldCreate = cls.create;
+      var axiom = this;
+
+      cls.create = function(args, X) {
+        // Install our own CSS, and then all parent models as well.
+        var c = this;
+        while ( c && c.private_ ) {
+          if ( X.installedCSS[c.id] ) break;
```

Comment:
1. Suppose I did have two or three `CSS` axioms, and each decorated `create` to install only itself. The outermost one works fine, but then an inner one would see that this class has already been installed in the document, and not install further CSS. It seemed simpler to combine multiple CSS axioms into a single install step.

2. I can make the once-per-document logic less CSS-specific. The `Window` can have `isClassInstalled()` and `installClass()` methods, rather than the specific mechanism.